gui/choices.py

```diff
@@ -177,76 +174,19 @@
     ('SMB3_11', _('SMB3_11')),
 )
 
-DOSCHARSET_CHOICES = (
-    'CP437',
-    'CP850',
-    'CP852',
-    'CP866',
-    'CP932',
-    'CP949',
-    'CP950',
-    'CP1026',
-    'CP1251',
-    'ASCII',
-)
-
-UNIXCHARSET_CHOICES = (
-    'UTF-8',
-    'ISO-8859-1',
-    'ISO-8859-15',
-    'GB2312',
-    'EUC-JP',
-    'ASCII',
-)
-
 
-class CHARSET(object):
-
-    __CODEPAGE = re.compile("(?P<name>CP|GB|ISO-8859-|UTF-)(?P<num>\d+)").match
-
-    __canonical = {'UTF-8', 'ASCII', 'GB2312', 'HZ-GB-2312', 'CP1361'}
-
-    def __check_codec(self, encoding):
-        try:
-            if codecs.lookup(encoding):
-                return encoding.upper()
-        except LookupError:
-            pass
-        return
+def UNIXCHARSET_CHOICES():
+    with client as c:
```

Comment:
What if connection fails? It will happen during build of FreeNAS. We need to have a fallback of default options.